src/main/scala/temple/generate/language/service/go/GoServiceGenerator.scala

```diff
@@ -96,22 +106,71 @@ object GoServiceGenerator extends ServiceGenerator {
     sb.toString
   }
 
+  private def generateJsonMiddleware(): String =
+    readFile("src/main/scala/temple/generate/language/service/go/genFiles/JsonMiddleware.go")
+
+  private def generateHandler(serviceName: String, endpoint: Endpoint): String =
+    s"func ${serviceName + endpoint}Handler(w http.ResponseWriter, r *http.Request) {}\n\n"
+
+  private def generateHandlers(serviceName: String, endpoints: Set[Endpoint]): String = {
+    val sb = new StringBuilder
+    if (endpoints.contains(Endpoint.ReadAll)) {
+      sb.append(generateHandler(serviceName, Endpoint.ReadAll))
+    }
+    if (endpoints.contains(Endpoint.Create)) {
+      sb.append(generateHandler(serviceName, Endpoint.Create))
+    }
+    if (endpoints.contains(Endpoint.Read)) {
+      sb.append(generateHandler(serviceName, Endpoint.Read))
+    }
+    if (endpoints.contains(Endpoint.Update)) {
+      sb.append(generateHandler(serviceName, Endpoint.Update))
+    }
+    if (endpoints.contains(Endpoint.Delete)) {
+      sb.append(generateHandler(serviceName, Endpoint.Delete))
+    }
+    sb.toString
+  }
+
+  private def generateErrors(serviceName: String): String =
+    Seq(
+      "package dao",
+      "",
+      """import "fmt"""",
+      "",
+      s"// Err${serviceName.capitalize}NotFound is returned when a ${serviceName} for the provided ID was not found",
+      s"type Err${serviceName.capitalize}NotFound int64",
+      "",
+      s"func (e Err${serviceName.capitalize}NotFound) Error() string " + CodeWrap.curly
+        .tabbed(s"""return fmt.Sprintf("${serviceName} not found with ID %d", e)""") + "\n",
+    ).mkString("\n")
+
   override def generate(serviceRoot: ServiceRoot): Map[File, FileContent] = {
     val usesComms = serviceRoot.comms.nonEmpty
     val serviceString = generatePackage("main") + generateImports(
         serviceRoot.name,
         serviceRoot.module,
         usesComms,
+      ) + generateGlobals(
+        serviceRoot.name,
+        usesComms,
       ) + generateMain(
         serviceRoot.name,
         usesComms,
         serviceRoot.endpoints,
         serviceRoot.port,
-      )
-    Map(File(serviceRoot.name, s"${serviceRoot.name}.go") -> serviceString)
+      ) + generateJsonMiddleware() + generateHandlers(
+        serviceRoot.name,
+        serviceRoot.endpoints,
+      ).stripLineEnd
```

Comment:
A little confused as to why we strip the trailing newline here: should these functions return trailing newlines at all?